File: go-sdk/pkg/models/operations/getretentionpoliciesid.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type GetRetentionPoliciesIDPathParams struct {
	RetentionPolicyID string `pathParam:"style=simple,explode=false,name=retention_policy_id"`
}

type GetRetentionPoliciesIDQueryParams struct {
	Fields []string `queryParam:"style=form,explode=false,name=fields"`
}

type GetRetentionPoliciesIDRequest struct {
	Retries     *utils.RetryConfig
	PathParams  GetRetentionPoliciesIDPathParams
	QueryParams GetRetentionPoliciesIDQueryParams
}

type GetRetentionPoliciesIDResponse struct {
	ClientError     *shared.ClientError
	ContentType     string
	RetentionPolicy *shared.RetentionPolicy
	StatusCode      int64
}

File: ts-sdk/src/sdk/models/operations/getretentionpoliciesid.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class GetRetentionPoliciesIdPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=retention_policy_id" })
  retentionPolicyId: string;
}


export class GetRetentionPoliciesIdQueryParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "queryParam, style=form;explode=false;name=fields" })
  fields?: string[];
}


export class GetRetentionPoliciesIdRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: GetRetentionPoliciesIdPathParams;

  @SpeakeasyMetadata()
  queryParams: GetRetentionPoliciesIdQueryParams;
}


export class GetRetentionPoliciesIdResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  retentionPolicy?: shared.RetentionPolicy;

  @SpeakeasyMetadata()
  statusCode: number;
}
